File: nicodegroot/src/nl/nicasso/ql/ast/statement/Statement.java
package nl.nicasso.ql.ast.statement;

import nl.nicasso.ql.ast.ASTNode;
import nl.nicasso.ql.ast.CodeLocation;
import nl.nicasso.ql.visitor.StatementVisitor;

public abstract class Statement extends ASTNode {

	public Statement(CodeLocation location) {
		super(location);
	}
	
	public abstract <T> T accept(StatementVisitor<T> visitor);

}

File: JordyHeemskerk/QL/QL/Statement.swift
//
//  Statement.swift
//  QL
//
//  Created by Jordy Heemskerk on 24/02/16.
//  Copyright © 2016 Jordy Heemskerk. All rights reserved.
//

import Foundation

protocol Statement: ASTNode {
    
    func accept<T: StatementVisitor>(visitor: T) -> T.StatementReturnType
    
}